astropy/coordinates/representation.py

```diff
@@ -869,6 +869,38 @@ def represent_as(self, other_class, differential_class=None):
 
             return new_rep
 
+    def transform(self, matrix):
+        r"""Transform coordinates using a 3x3 matrix in a Cartesian basis.
+
+        This returns a new representation and does not modify the original one.
+        Any differentials attached to this representation will also be
+        transformed.
+
+        Parameters
+        ----------
+        matrix : (3,3) `~numpy.ndarray`
+            A 3x3 transformation matrix, such as a rotation matrix.
+
+        Notes
+        -----
+        In a Cartesian basis, a matrix is a rotation if and only if it is
+        orthogonal (:math:`M M^T = I`) and has determinant one
+        (:math:`det(M) = 1`)
+
+        """
+        # route transformation through Cartesian
+        difs_cls = {k: CartesianDifferential for k in self.differentials.keys()}
+        crep = self.represent_as(CartesianRepresentation,
+                                 differential_class=difs_cls
+                                ).transform(matrix)
+
+        # move back to original representation
+        difs_cls = {k: diff.__class__ for k, diff in self.differentials.items()}
+        rep = crep.represent_as(self.__class__, difs_cls)
+        # TODO, get units to be the same as input units
```

Comment:
Let's not worry about that level of detail - at least not in the low-level classes.

Reply:
I just know I could. The PhysicSpherical case was easy because of how Quantity works, I was thinking of something like

```
For comp, unit in rep._units.items():
    if unit.physical_type == "angle":
        original_unit = getattr(self, comp).unit
        getattr(rep, comp) << original_unit
        rep._units[comp] = original_unit
```

Reply:
I got it to work, but I don't think I'm going to include it. I agree it's probably not necessary for these classes, but it might be nice...

```
# Get units to be the same as input units
for c, unit in rep._units.items():
    if unit.physical_type == "angle":
        original_unit = getattr(self, c).unit
        setattr(rep, "_" + c, getattr(rep, c) << original_unit)
        rep._units[c] = original_unit
```

Reply:
Yes, definitely possible. But for many applications speed will be more important... Let's get a `.to(...)` method in another PR!